### ISSUE TITLE
Library build issue on macOS

### ISSUE THREAD

**Issue by byadav-ciena** [zmij/afsm#21]
Cloned the library to Mac and went through the build. It has compile errors in your support library. Full log is below.

> cmake ..
-- The C compiler identification is AppleClang 11.0.0.11000033
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at CMakeLists.txt:38 (message):
  ccache enabled, but not found

-- AFSM C++ standard 11
Cloning into 'cmake-scripts'...
-- Metapushkin include dir /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/<username>/Projects/FSM/afsm/build
<username>@ONM-BYADAV-02:~/Projects/FSM/afsm/build$ make
Scanning dependencies of target gtest
[  2%] Creating directories for 'gtest'
[  5%] Performing download step (git clone) for 'gtest'
-- gtest download command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/gtest-stamp/gtest-download-*.log
[  8%] No patch step for 'gtest'
[ 11%] No update step for 'gtest'
[ 14%] Performing configure step for 'gtest'
-- gtest configure command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/gtest-stamp/gtest-configure-*.log
[ 17%] Performing build step for 'gtest'
-- gtest build command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/gtest-stamp/gtest-build-*.log
[ 20%] No install step for 'gtest'
[ 22%] Completed 'gtest'
[ 22%] Built target gtest
Scanning dependencies of target gmock
[ 25%] Creating directories for 'gmock'
[ 28%] Performing download step (git clone) for 'gmock'
-- gmock download command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/gmock-stamp/gmock-download-*.log
[ 31%] No patch step for 'gmock'
[ 34%] No update step for 'gmock'
[ 37%] Performing configure step for 'gmock'
-- gmock configure command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/gmock-stamp/gmock-configure-*.log
[ 40%] Performing build step for 'gmock'
-- gmock build command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/gmock-stamp/gmock-build-*.log
[ 42%] No install step for 'gmock'
[ 45%] Completed 'gmock'
[ 45%] Built target gmock
Scanning dependencies of target metapushkin
[ 48%] Creating directories for 'metapushkin'
[ 51%] Performing download step (git clone) for 'metapushkin'
-- metapushkin download command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/metapushkin-stamp/metapushkin-download-*.log
[ 54%] No patch step for 'metapushkin'
[ 57%] No update step for 'metapushkin'
[ 60%] Performing configure step for 'metapushkin'
-- metapushkin configure command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/metapushkin-stamp/metapushkin-configure-*.log
[ 62%] Performing build step for 'metapushkin'
-- metapushkin build command succeeded.  See also /Users/<username>/Projects/FSM/afsm/build/src/metapushkin-stamp/metapushkin-build-*.log
[ 65%] No install step for 'metapushkin'
[ 68%] Completed 'metapushkin'
[ 68%] Built target metapushkin
Scanning dependencies of target test-afsm-base
[ 71%] Building CXX object test/CMakeFiles/test-afsm-base.dir/__/lib/ansi-colors/src/ansi_colors.cpp.o
[ 74%] Building CXX object test/CMakeFiles/test-afsm-base.dir/static_tests.cpp.o
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:37:21: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct join< ::std::integer_sequence<T, A...>, ::std::integer_sequence<T, B...> > {
             ~~~~~~~^~~~~~~~~~~~~~~~
                    __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:37:55: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct join< ::std::integer_sequence<T, A...>, ::std::integer_sequence<T, B...> > {
                                               ~~~~~~~^~~~~~~~~~~~~~~~
                                                      __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:38:25: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
    using type = ::std::integer_sequence<T, A..., B...>;
                 ~~~~~~~^~~~~~~~~~~~~~~~
                        __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:42:21: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct join< ::std::integer_sequence<T, A...>, ::std::integral_constant<T, B > > {
             ~~~~~~~^~~~~~~~~~~~~~~~
                    __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:43:25: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
    using type = ::std::integer_sequence<T, A..., B>;
                 ~~~~~~~^~~~~~~~~~~~~~~~
                        __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:47:53: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct join< ::std::integral_constant<T, A>, ::std::integer_sequence<T, B...> > {
                                             ~~~~~~~^~~~~~~~~~~~~~~~
                                                    __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:48:25: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
    using type = ::std::integer_sequence<T, A, B...>;
                 ~~~~~~~^~~~~~~~~~~~~~~~
                        __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:61:25: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
    using type = ::std::integer_sequence< T, V, Values... >;
                 ~~~~~~~^~~~~~~~~~~~~~~~
                        __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:72:25: error: no template
      named 'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
    using type = ::std::integer_sequence< T, V, Values... >;
                 ~~~~~~~^~~~~~~~~~~~~~~~
                        __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:19:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:149:20: error: no member
      named 'index_sequence' in namespace 'std'
            ::std::index_sequence<Head>,
            ~~~~~~~^
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:149:40: error: expected
      class name
            ::std::index_sequence<Head>,
                                       ^
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:150:20: error: unknown
      template name 'index_sequence'
            ::std::index_sequence<>
                   ^
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/integer_sequence.hpp:163:25: error: no template
      named 'index_sequence' in namespace 'std'
    using type = ::std::index_sequence<>;
                 ~~~~~~~^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:47:16: error: variable templates
      are a C++14 extension [-Werror,-Wc++14-extensions]
constexpr bool contains_v = contains_t<T, Y...>::value;
               ^
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:68:24: error: no template named
      'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct is_empty<::std::integer_sequence<T>> : ::std::true_type {};
                ~~~~~~~^~~~~~~~~~~~~~~~
                       __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:70:24: error: no template named
      'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct is_empty<::std::integer_sequence<T, Values...>> : ::std::false_type {};
                ~~~~~~~^~~~~~~~~~~~~~~~
                       __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:75:16: error: variable templates
      are a C++14 extension [-Werror,-Wc++14-extensions]
constexpr bool is_empty_v = is_empty_t<T...>::value;
               ^
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:102:21: error: no template named
      'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct front<::std::integer_sequence<T, V, Values...>> : std::integral_constant<T, V> {};
             ~~~~~~~^~~~~~~~~~~~~~~~
                    __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
In file included from /Users/<username>/Projects/FSM/afsm/test/static_tests.cpp:9:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/fsm.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/detail/base_states.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/include/afsm/definition.hpp:11:
In file included from /Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta.hpp:14:
/Users/<username>/Projects/FSM/afsm/build/src/metapushkin/include/pushkin/meta/algorithm.hpp:104:21: error: no template named
      'integer_sequence' in namespace 'std'; did you mean '__integer_sequence'?
struct front<::std::integer_sequence<T>> {
             ~~~~~~~^~~~~~~~~~~~~~~~
                    __integer_sequence
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:93:8: note:
      '__integer_sequence' declared here
struct __integer_sequence {
       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [test/CMakeFiles/test-afsm-base.dir/static_tests.cpp.o] Error 1
make[1]: *** [test/CMakeFiles/test-afsm-base.dir/all] Error 2
make: *** [all] Error 2

**Comment by zmij** [comment]
Oh, I'm really very sorry. The develop branch is somewhat stale, and it requires C++14, will fix today.

**Comment by zmij** [comment]
Fixed by 49181bf52fa2ab8bcea6017d11b3cd321b56c13c
